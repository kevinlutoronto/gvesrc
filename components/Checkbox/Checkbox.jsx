// @flow
import React from 'react';
import cx from 'classnames';
import style from './style.scss';
import Icon from '../Icon/Icon';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';

type Props = {|
  id?: string,
  checked?: boolean,
  data?: Data,
  description?: string,
  disabled?: boolean,
  fancy?: boolean,
  icon?: string,
  indeterminate?: boolean,
  isBold?: boolean,
  name?: string,
  onBlur?: () => mixed,
  onChange?: () => mixed,
  text: React$Node,
  value?: boolean,
|};

class Checkbox extends React.Component<Props> {
  static defaultProps = {
    checked: false,
    indeterminate: false,
    name: 'checkbox',
    text: 'Option',
    value: false,
  };

  render() {
    const {
      checked,
      data,
      description,
      disabled,
      fancy,
      icon,
      indeterminate,
      isBold,
      name,
      onBlur,
      onChange,
      text,
      value,
      id,
    } = this.props;

    const isChecked = checked || value;
    const inlineClass = cx({ [style.inline]: fancy });
    const inputClassName = indeterminate ? style.indeterminate : style.original;
    const wrapperClassName = cx(style.wrapper, {
      [style.wrapperDisabled]: disabled,
      [style.fancy]: fancy,
      [style.fancyChecked]: fancy && isChecked,
    });
    const textClassName = cx(style.text, {
      [style.textBold]: isBold,
      [style.textDisabled]: disabled,
    });

    return (
      <div {...getDataAttrs(data)} className={inlineClass}>
        <div>
          <label htmlFor={id} className={wrapperClassName}>
            <input
              checked={isChecked}
              className={inputClassName}
              disabled={disabled}
              id={id}
              name={name}
              onChange={onChange}
              onBlur={onBlur}
              type="checkbox"
            />
            {!fancy && <div className={style.checkbox} />}
            {icon && (
              <div className={fancy ? style.fancyIcon : style.icon}>
                <Icon icon={icon} />
              </div>
            )}
            <div className={textClassName}>{text}</div>
          </label>
        </div>
        {description && <span className={style.description}>{description}</span>}
      </div>
    );
  }
}

Checkbox.displayName = 'Plasma@Checkbox';

export default Checkbox;
