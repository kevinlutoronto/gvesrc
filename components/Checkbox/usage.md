### Usage
- Used when there are multiple choices, and more than one option can be selected.
- Consider using a multi-select if there are several(+) choices — something to consider if your form is long, or space is limited.
- Checkboxes are preferable over a multi-select when it’s important that the user sees the options available to them, to understand and consider, versus a more passive scan of their options via a multi-select.
