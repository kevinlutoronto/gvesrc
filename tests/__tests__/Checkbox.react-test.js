/* eslint-disable */

import React from 'react';
import Checkbox from '../../components/Checkbox/Checkbox';
import renderer from 'react-test-renderer';

describe('Checkbox Component', () => {
  test('render', () => {
    const component = renderer.create(<Checkbox />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.checked = true;
    expect(tree).toMatchSnapshot();
  });

  test('render', () => {
    const component = renderer.create(<Checkbox fancy />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
