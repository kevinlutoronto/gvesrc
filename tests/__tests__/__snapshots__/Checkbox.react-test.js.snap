// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component render 1`] = `
<div
  className=""
>
  <div>
    <label
      className="wrapper"
      htmlFor={undefined}
    >
      <input
        checked={false}
        className="original"
        disabled={undefined}
        id={undefined}
        name="checkbox"
        onBlur={undefined}
        onChange={undefined}
        type="checkbox"
      />
      <div
        className="checkbox"
      />
      <div
        className="text"
      >
        Option
      </div>
    </label>
  </div>
</div>
`;

exports[`Checkbox Component render 2`] = `
<div
  checked={true}
  className=""
>
  <div>
    <label
      className="wrapper"
      htmlFor={undefined}
    >
      <input
        checked={false}
        className="original"
        disabled={undefined}
        id={undefined}
        name="checkbox"
        onBlur={undefined}
        onChange={undefined}
        type="checkbox"
      />
      <div
        className="checkbox"
      />
      <div
        className="text"
      >
        Option
      </div>
    </label>
  </div>
</div>
`;

exports[`Checkbox Component render 3`] = `
<div
  className="inline"
>
  <div>
    <label
      className="wrapper fancy"
      htmlFor={undefined}
    >
      <input
        checked={false}
        className="original"
        disabled={undefined}
        id={undefined}
        name="checkbox"
        onBlur={undefined}
        onChange={undefined}
        type="checkbox"
      />
      <div
        className="text"
      >
        Option
      </div>
    </label>
  </div>
</div>
`;
